src/merge_bed.py

```diff
@@ -28,10 +30,41 @@ def read_bed_file(f: TextIO) -> list[BedLine]:
 
     return res
 
+def is_BEDline_before (el1:BedLine, el2:BedLine) -> bool:
+    """check if a BEDline is before another
+    >>> is_BEDline_before(BedLine("chr1", 1, 2, "foo"), BedLine("chr1", 5, 6, "foo"))
+    True
+    """
+    for i in ['chrom', 'chrom_start', 'chrom_end']:
+        if getattr(el1,i) > getattr(el2,i):
+            return False
+    return True 
 
-def merge(f1: list[BedLine], f2: list[BedLine], outfile: TextIO) -> None:
+def merge_sort_generator (list_1: list[BedLine], list_2: list[BedLine]) -> Generator [BedLine, None, None ]:
+    """merges 2 sorted lists and gives elements."""
+    iter1,iter2= iter(list_1), iter(list_2)
+    for (el1, el2) in zip(iter1,iter2): 
+        if is_BEDline_before (el1,el2):
+            yield el1
+        yield el2 
+    for el1 in iter1: 
+        yield el1 
+    for el2 in iter2: 
+        yield el2 
```

Comment:
I am surprised that this made it through the tests. I shouldn't.

When you take two sequences and zip them, you get their elements paired up (to the length of the shortest sequence).

So, if you have

```
x = [1, 2, 3, 4]
y = [7, 8, 9]
```

and you zip them, you get

```
(1,7)
(2,8)
(3,9)
```

which, if you order them pairwise as you do will be

```
1, 7, 2, 8, 3, 9
```

which isn't right.